Source/Core/UICommon/GameFile.cpp

```diff
@@ -210,11 +224,61 @@ bool GameFile::BannerChanged()
   return !m_pending.volume_banner.buffer.empty();
 }
 
-void GameFile::BannerCommit()
+void GameFile::WiiBannerCommit()
 {
   m_volume_banner = std::move(m_pending.volume_banner);
 }
 
+bool GameFile::ReadPNGBanner(const std::string& path)
+{
+  File::IOFile file(path, "rb");
+  if (!file)
+    return false;
+
+  std::vector<u8> png_data(file.GetSize());
+  if (!file.ReadBytes(png_data.data(), png_data.size()))
+    return false;
+
+  GameBanner& banner = m_pending.custom_banner;
+  std::vector<u8> data_out;
+  if (!Common::LoadPNG(png_data, &data_out, &banner.width, &banner.height))
+    return false;
+
+  banner.buffer.resize(data_out.size() / sizeof(u32));
+  for (size_t i = 0; i < banner.buffer.size(); i++)
```

Comment:
What does this loop do? From the pattern below I'm guessing something like RGBA to ARGB (or whatever) - I'd personally like a comment pointing that out (plus, "why"); feel free to ignore this comment if I'm the only one.

Reply:
I've added a comment. There isn't much of a "why", though – we just happen to be using different formats in the different parts of the code.